api/Wrapper.ts

```diff
@@ -12,26 +12,46 @@ const api = axios.create({
  * @param city_name The name of the city to create the survey for
  * @param onFail A callback to do any desired error handling
  */
-export const createSurvey = async (
-  city_name?: string,
+export const request = async (
+  config: AxiosRequestConfig,
   onFail: (e: any) => void = console.log
 ) => {
   try {
-    const response = await api.get("/api/create/survey", {
-      params: {
-        city_name: city_name,
-      },
-    });
-
-    if (!_.isUndefined(response.data) && !_.isNull(response.data)) {
-      const surveyId: string = response.data.id;
-      return surveyId;
+    const response = await api.request(config);
+    if (!_.isUndefined(response.data)) {
+      return response.data;
     } else {
-      onFail("");
       return null;
     }
   } catch (e) {
     onFail(e);
     return null;
   }
 };
+
+/**
+ * Create a survey and return the survey ID.
+ *
+ * @param city_name The name of the city to create the survey for
+ * @param onFail A callback to do any desired error handling
+ */
+export const createSurvey = async (
+  city_name?: string,
+  onFail: (e: any) => void = console.log
```

Comment:
We should add a TODO here so we don't forget to handle this exception later (display an error message to the user, etc).

Reply:
What do you mean by "handle this exception"? This is just a callback parameter so the calling code can do the error handling however it wants to. It just defaults to calling `console.log` on the error if no error handling callback was provided

Reply:
Oh I see, I misunderstood the code then, we should just make sure we have proper error handling, that is all I was trying to say.